subsystem/kubeapi_test.go

```diff
@@ -0,0 +1,305 @@
+			Kind:       "Image",
+			APIVersion: getAPIVersion(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: Options.Namespace,
+			Name:      spec.ClusterRef.Name,
+		},
+		Spec: *spec,
+	})
+	Expect(err).ShouldNot(HaveOccurred())
+}
+
+func waitForImageCRDState(
+	ctx context.Context, client k8sclient.Client, key types.NamespacedName, message string, timeout int) {
+
+	image := &v1alpha1.Image{}
+	start := time.Now()
+	for time.Duration(timeout)*time.Second > time.Since(start) {
+		image = getImageCRD(ctx, client, key)
+		if image.Status.State != "" {
+			Expect(image.Status.State).Should(Equal(message))
+		}
+		time.Sleep(time.Second)
+	}
+	Expect(image.Status.State).Should(Equal(message))
+}
+
+func getImageCRD(ctx context.Context, client k8sclient.Client, key types.NamespacedName) *v1alpha1.Image {
+	image := &v1alpha1.Image{}
+	err := client.Get(ctx, key, image)
+	Expect(err).ShouldNot(HaveOccurred())
+	return image
+}
+
+func cleanUP(ctx context.Context, client k8sclient.Client) {
+	Expect(client.DeleteAllOf(ctx, &v1alpha1.Cluster{}, k8sclient.InNamespace(Options.Namespace))).ShouldNot(HaveOccurred())
+	Expect(client.DeleteAllOf(ctx, &v1alpha1.Image{}, k8sclient.InNamespace(Options.Namespace))).ShouldNot(HaveOccurred())
+}
+
+var _ = Describe("cluster installation", func() {
+	ctx := context.Background()
+
+	waitForClusterReconcileTimeout := 10
+
+	secretRef := &corev1.SecretReference{}
+
+	generateFAPostStepReply := func(h *models.Host, freeAddresses models.FreeNetworksAddresses) {
+		fa, err := json.Marshal(&freeAddresses)
+		Expect(err).NotTo(HaveOccurred())
+		_, err = agentBMClient.Installer.PostStepReply(ctx, &installer.PostStepReplyParams{
+			ClusterID: h.ClusterID,
+			HostID:    *h.ID,
+			Reply: &models.StepReply{
+				ExitCode: 0,
+				Output:   string(fa),
+				StepID:   string(models.StepTypeFreeNetworkAddresses),
+				StepType: models.StepTypeFreeNetworkAddresses,
+			},
+		})
+		Expect(err).ShouldNot(HaveOccurred())
+	}
+
+	BeforeSuite(func() {
+		if !Options.EnableKubeAPI {
+			return
+		}
+		secretRef = &corev1.SecretReference{
+			Namespace: Options.Namespace,
+			Name:      "pull-secret",
+		}
+		err := kubeClient.Get(ctx, types.NamespacedName{Namespace: Options.Namespace, Name: "pull-secret"}, &corev1.Secret{})
+		if apierrors.IsNotFound(err) {
+			deployPullSecretResource(ctx, kubeClient, secretRef.Name, pullSecret)
+		} else {
+			Expect(err).ShouldNot(HaveOccurred())
+		}
+		cleanUP(ctx, kubeClient)
+	})
+	AfterSuite(func() {
+		if !Options.EnableKubeAPI {
+			return
+		}
+	})
+
+	Context("cluster", func() {
+		AfterEach(func() {
+			cleanUP(ctx, kubeClient)
+			clearDB()
+		})
+
+		It("deploy cluster and start installation", func() {
+			spec := getDefaultClusterSpec(secretRef)
+			deployClusterCRD(ctx, kubeClient, spec)
+			key := types.NamespacedName{
+				Namespace: Options.Namespace,
+				Name:      spec.Name,
+			}
+			cluster := getClusterFromDB(ctx, kubeClient, db, key, waitForClusterReconcileTimeout)
```

Comment:
You can move this part to the end of the test, then you will not need to handle the update issue.